core/dbt/events/base_types.py

```diff
@@ -88,6 +102,18 @@ def get_pid(self) -> int:
             self.pid = os.getpid()
         return self.pid
 
+    def get_node_info(self):
+        return None
+
+
+class NodeInfo(Event, metaclass=ABCMeta):
+    report_node_data: Any
+    node_status: str
+    run_state: str
+
+    def get_node_info(self):
+        return vars(Node(self.report_node_data))
```

Comment:
So this is straying from the nominal typing we're using in most of the event module to use structural typing instead. If we don't have a way to explicitly describe the fields we care about returning to users, this is ok for now, but it means that if we make a change on the node types, it might blow our logging interface and break downstream consumers without us really knowing.

Reply:
I'm not following.  What would you expect here?